packages/drivers/odsp-driver/src/zipItDataRepresentationUtils.ts

```diff
@@ -0,0 +1,496 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert, IsoBuffer, Uint8ArrayToString } from "@fluidframework/common-utils";
+
+/** Default encoding for strings */
+const stringEncoding = "utf-8";
+
+/**
+ * Buffer class, used to sequentially read data.
+ * Used by tree code to reconstruct a tree from binary representation.
+ */
+export class ReadBuffer {
+    protected index = 0;
+
+    public get buffer() {
+        return this.data;
+    }
+
+    constructor(protected readonly data: Uint8Array) {
+        // BlobShallowCopy will return to users parts of this array.
+        // We need to ensure that nobody can change it, as it will have
+        // catastrophic result and will be really hard to investigate.
+        Object.freeze(data.buffer);
+    }
+
+    public get eof() { return this.index === this.data.length; }
+    public get pos() { return this.index; }
+    public get length() { return this.data.length; }
+
+    public slice(start: number, end: number) {
+        return this.data.slice(start, end);
+    }
+
+    public reset() {
+        this.index = 0;
+    }
+
+    public read(lengthArg = 1): number {
+        let res = 0;
+        let multiplier = 1;
+        let length = lengthArg;
+        while (length > 0) {
+            assert(!this.eof, "unexpected end of buffer");
+            res += this.data[this.index] * multiplier;
+            this.index++;
+            multiplier *= 256;
+            length--;
+        }
+        return res;
+    }
+
+    public skip(length: number) {
+        assert(length >= 0, "Skip length should be positive");
+        this.index += length;
+    }
+}
+
+/**
+ * Buffer class, used to sequentially writ data.
+ * Used by tree code to serialize tree into binary representation.
+ */
+class WriteBuffer {
```

Comment:
Please take a look if there is reasonable way to partition code and separate writing code into separate .ts file.
That way tree shaking can eliminate it as we will not use it (only in tests), but tree shaking is dump and pulls whole file.
It might be tricky as it's not just one class, but methods like BlobCore.write(), TreeBuilder.serialize().
If possible, we should partition these classes into read-only versions, and extensions (possibly stand-alone functions) that provide writing capabilities. It might be challenging, as it might require more things to be made public :(

Reply:
That said, it's possible that it's Ok to leave it as is with assumption that at some point in future we would want to move summarization process to use similar more compact format, and thus writing capabilities would be required in shipping. So we should do splitting if it makes sense / feels natural, otherwise get a bit more data.

Reply:
I did put the read and write in different files just for separation of concerns. Otherwise we do inter dependent in a lot of things. Also those seems to be a small code, so it should not affect that much.